### DISCUSSION TITLE
「重学TS 2.0 」TS 面试题：泛型是什么，它有什么作用？

### DISCUSSION THREAD

**Discussion by semlinker** [semlinker/awesome-typescript#75] [General]
### 在 TypeScript 中泛型是什么，它有什么作用？

**Comment by wenkm** [comment]
释义：泛型 -> 宽泛的类型。
用法：在定义类、接口、函数的时候一起服用
作用：在定义的时候不指定类型，使用的时候才定义类型的这么一个玩意儿
准字：业界不成文规定 `<T>`
副作用：可以继承

**Comment by douhao1988** [comment]
- 泛型是一种创建可复用代码块的工具。这类代码块不只能被一种类型使用，而是能被多种类型复用。  
- 类似于参数的作用，泛型是一种用以增强类（classes）、类型（types）和接口（interfaces）能力的非常可靠的手段。  
- 这样，我们开发者，就可以轻松地将那些可复用的代码块，适用于各种输入。
